src/SDK/Trace/Behavior/SpanExporterTrait.php

```diff
@@ -44,7 +56,87 @@ public function export(iterable $spans): int
             return SpanExporterInterface::STATUS_SUCCESS;
         }
 
-        return $this->doExport($spans); /** @phpstan-ignore-line */
+        // If retryPolicy is not set then just doExport once and return the status
+        if ($this->retryPolicy === null) {
+            self::logInfo('retry polict is null. Exporting the spans without retry' . PHP_EOL);
+
+            return $this->doExport($spans); /** @phpstan-ignore-line */
+        }
+
+        while ($shouldRetry) {
```

Comment:
This adds a lot to the cyclomatic complexity of this method.
Could you extract the logic into a separate method to make it easier to follow?

Reply:
@tidal:
It is just a while loop for no of attempts allowed. 
 Can you give me some more idea about how I can improve it?

Reply:
I did not mean to improve the loop itself, but maybe to simply extract it.
However it's not be super important and could be done in a refactoring later.